app/src/main/java/org/thoughtcrime/securesms/repository/ConversationRepository.kt

```diff
@@ -180,59 +240,83 @@ class DefaultConversationRepository @Inject constructor(
         storage.setRecipientApproved(recipient, isApproved)
     }
 
-    override suspend fun deleteForEveryone(
+    override suspend fun deleteCommunityMessagesRemotely(
         threadId: Long,
-        recipient: Recipient,
-        message: MessageRecord
-    ): Result<Unit> = suspendCoroutine { continuation ->
-        buildUnsendRequest(recipient, message)?.let { unsendRequest ->
-            MessageSender.send(unsendRequest, recipient.address)
+        messages: Set<MessageRecord>
+    ) {
+        val community = lokiThreadDb.getOpenGroupChat(threadId) ?:
+            throw Exception("Not a Community")
```

Comment:
Just an FYI there's a nicer interface for this:
```suggestion
        val community = checkNotNull(lokiThreadDb.getOpenGroupChat(threadId)) { "Not a community" }
```

Nothing wrong with your approach though, just in case you haven't used this before.

Reply:
Nice. Changed it.